alerts_backend/python/app.py

```diff
@@ -45,45 +46,105 @@
-        # Create the table if it doesn't exist
+        # Create the table(s) if they don't exist
         metadata_obj.create_all(engine)
-        app.logger.info("Table successfully created (if not already created)")
+        app.logger.info("Table(s) successfully created (if not already created)")
     except exc.SQLAlchemyError as e:
-        # Since creation of table is a critical error, raise exception
-        app.logger.error(f"SQL error occurred during creation of table (CRITICAL - THROWING ERROR): {e}")
+        # Since creation of table(s) is a critical error, raise exception
+        app.logger.error(f"SQL error occurred during creation of table(s) (CRITICAL - THROWING ERROR): {e}")
         raise e
 
 
-def insert_into_db(data_to_insert: Dict[str, Union[str, int, bool]]) -> int:
+def insert_into_table(data_to_insert: Dict[str, Union[str, int, bool]], table: Table) -> int:
     """
-    Insert object into the database based off of the engine.
-    Assumes data_to_insert has values for all the keys:
-    fa_alert_id, ident, origin, destination, aircraft_type, start_date, end_date.
+    Insert object into the database based off of the table.
+    Assumes data_to_insert has values for all the keys
+    that are in the data_to_insert variable, and also that
+    table is a valid SQLAlchemy Table variable inside the database.
     Returns 0 on success, -1 otherwise
     """
     try:
         with engine.connect() as conn:
-            stmt = insert(aeroapi_alert_configurations)
+            stmt = insert(table)
             conn.execute(stmt, data_to_insert)
             conn.commit()
-
-            app.logger.info("Data successfully inserted into table")
-
+            app.logger.info(f"Data successfully inserted into table {table.name}")
     except exc.SQLAlchemyError as e:
-        app.logger.error(f"SQL error occurred during insertion into table: {e}")
+        app.logger.error(f"SQL error occurred during insertion into table {table.name}: {e}")
         return -1
-
     return 0
 
 
+@app.route("/post", methods=["POST"])
+def handle_alert() -> (Response, int):
+    """
+    Function to receive AeroAPI POST requests. Filters the request
+    and puts the necessary data into the SQL database.
+    Returns a JSON Response and also the status code in a tuple.
+    """
+    # Form response
+    r_title: str
+    r_detail: str
+    r_status: int
+    data: Dict[Any] = request.json
+    # Process data by getting things needed
+    # If value doesn't exist, default to None
+    processed_data: Dict[Any] = dict()
+    processed_data["long_description"] = data.get("long_description", None)
+    processed_data["short_description"] = data.get("short_description", None)
+    processed_data["summary"] = data.get("summary", None)
+    processed_data["event_code"] = data.get("event_code", None)
+    processed_data["alert_id"] = data.get("alert_id", None)
+    processed_data["fa_flight_id"] = data.get("flight", None).get("fa_flight_id", None)
```

Comment:
This will blow up if `flight` is missing since you'll then be calling `.get` on `None`. Generally, this block is a bit of an eyesore. What if you structured it more like so:
```
try:
    processed_date = {
        "long_description": data["long_description"],
        ...
        "fa_flight_id": data["flight"]["fa_flight_id"],
        ...
    }
except KeyError as e:
    # log an error
    # but still insert what we can into the database just in case FlightAware sent us anything useful whatsoever
   # still return 400, though

Reply:
Not sure how I could still insert what we can into the database if everything is inserted at once? Should I surround it with a while result is None or soemthing?

Reply:
eeehhh yeah you're right, we'll potentially be missing lots of flight data. Let's forget about inserting if there's required data missing from the payload.

Reply:
Oh I just did not know programming wise how to get it, like if some variable was None not sure how to tell which variable was None and thus replace it, but I'll assume no insertion is ok for now then